src/app.js

```diff
@@ -1,31 +1,35 @@
-'use strict';
-
-/* istanbul ignore next */ if (process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test')
-  require('dotenv').config(); // eslint-disable-line import/no-extraneous-dependencies
-
-const express = require('express');
+// eslint-disable-next-line import/no-extraneous-dependencies
+import dotenv from 'dotenv';
+import express from 'express';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+import Settings from './lib/settings.js';
+import * as addons from './lib/addons.js';
+import { logger } from './lib/utilities.js';
+import controllers from './controllers/index.js';
+
+/* istanbul ignore next */ if (['development', 'test'].includes(Settings.env)) dotenv.config();
 
 const app = express();
-const addons = require('./lib/addons');
-
-const { logger } = require('./lib/utilities');
 
-/* istanbul ignore next */ if (!global.__basedir) {
+/* istanbul ignore next */ if (!global?.__basedir) {
+  // eslint-disable-next-line no-global-assign
+  __dirname = dirname(fileURLToPath(import.meta.url));
   global.__basedir = __dirname;
 }
 
 // middleware
 app.use(express.json());
-addons.init(app);
+await addons.init(app);
 
 // actual api routes
 logger.info('Setting up routes...');
-app.use(require('./controllers'));
+app.use(controllers);
```

Comment:
Just a reminder since we have top level await, this is also posible, i don't have a preference btw performance impact should be negligible 
```js
app.use(await import('./controllers/index.js'));
```

Reply:
yeah, but i don't see the need for more async imports when we don't need to. before it was just convenience, i'd rather avoid more awaits